app/controllers/api/proxy_rules_controller.rb

```diff
@@ -44,7 +44,7 @@ def destroy
   private
 
   def authorize!
-    provider_can_use!(:independent_mapping_rules)
+    raise ActiveRecord::RecordNotFound unless service.independent_mapping_rules?
```

Comment:
Not sure about `RecordNotFound` here.

Also... instead of https://github.com/3scale/porta/blob/09fa0ef1cff253740ba44a8ee4929d7a1c789c86/app/controllers/api/proxy_rules_controller.rb#L54-L56 maybe we could use https://github.com/3scale/porta/blob/9529edd7d8d5058ea92ea8f2ee990e36bb51ee21/app/controllers/frontend_controller.rb#L111 with a `before_action`. WDYT?

Reply:
the RecordNotFound is to mimic the same behave that we had with `provider_can_use!`. wdyt?

good idea.. will move it to a before_action so